backend/src/mail_sender.rs

```diff
@@ -0,0 +1,146 @@
+use std::error::Error;
+use std::fmt::Display;
+use std::fs;
+
+use lettre::message::Mailbox;
+use lettre::message::{Attachment, Body, MultiPart, header::ContentType};
+use lettre::{Address, Message, SmtpTransport, Transport};
+
+#[derive(Debug)]
+pub enum MailSenderError {
+    InvalidEmail,
+    NotInDatabase,
+    InvalidFilePath,
+    NoReceivers,
+    NoFile,
+    CouldntSendEmail,
+}
+
+impl Error for MailSenderError {}
+
+impl Display for MailSenderError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            MailSenderError::InvalidEmail => write!(f, "invalid email"),
+            MailSenderError::NotInDatabase => write!(f, "not in database"),
+            MailSenderError::InvalidFilePath => write!(f, "invalid file path"),
+            MailSenderError::NoReceivers => write!(f, "no receivers"),
+            MailSenderError::NoFile => write!(f, "no file"),
+            MailSenderError::CouldntSendEmail => write!(f, "couldn't send email"),
+        }
+    }
+}
+
+#[derive(Debug)]
+
+pub struct Receiver {
+    pub name: String,
+    pub mail: Address,
+}
+
+pub struct MailSender {
+    people: Vec<Receiver>,
+    file: Body,
+}
+
+impl Default for MailSender {
+    fn default() -> Self {
+        MailSender {
+            people: vec![],
+            file: Body::new("".to_string()),
+        }
+    }
+}
+
+impl MailSender {
+    pub fn add_person(&mut self, person: Receiver) {
+        self.people.push(person);
+    }
+
+    pub fn remove_person(&mut self, person: Receiver) -> Result<(), MailSenderError> {
+        let wanted_mail = person.mail;
+
+        let person_position = self.people.iter().position(|x| x.mail == wanted_mail);
+
+        if person_position.is_none() {
+            return Err(MailSenderError::NotInDatabase);
+        }
+
+        self.people.remove(person_position.unwrap());
+
+        Ok(())
+    }
+
+    pub fn add_file(&mut self, path: &str) -> Result<(), MailSenderError> {
+        let pdf_file = fs::read(path);
+
+        if pdf_file.is_err() {
+            return Err(MailSenderError::InvalidFilePath);
+        }
+
+        self.file = Body::new(pdf_file.unwrap());
+
+        Ok(())
+    }
+
+    pub fn send(&self) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Budeš tu samou instanci email někdy posílat 2x? Jestli ne, tak je idiomatický brát jenom `self` (ne referenci) a rozebrat to ať nemusíš klonovat